src/pages/LogView/LoadingPage/useResolveLogURL.ts

```diff
@@ -74,12 +91,21 @@ export const useResolveLogURL = ({
         jobLogsURL = getJobLogsURL(buildID);
         legacyJobLogsURL = getLegacyJobLogsURL(buildID);
       }
+      downloadURL = rawLogURL;
       break;
     }
     case LogTypes.EVERGREEN_TASK_FILE: {
       if (taskID && execution && fileName) {
-        // TODO: resolve this value using GQL https://jira.mongodb.org/browse/EVG-20809
-        rawLogURL = getEvergreenTaskFileURL(taskID, execution, fileName);
+        downloadURL = getEvergreenTaskFileURL(
+          taskID,
+          execution,
+          encodeURIComponent(fileName)
+        );
+        const allFiles = taskFileData?.task?.files.groupedFiles.flatMap(
+          (group) => group.files
+        );
+        rawLogURL =
```

Comment:
So for downloading task files we want to use the evergreen proxy for downloading task files because it will allow us to get faster compressed download speeds, leading to a better user experience. But we don't want users to use that route for non-parsley downloads which is why the raw link (The one in the details dropdown) will link to the file directly in S3.  You can check out the [tech design](https://docs.google.com/document/d/10qJmO3lkMvUBpOApXu9ovy9m0ARBQGbaOvoFUbRW6FE/edit#heading=h.2s9zdv6a57gn) for more information.